src/opm/parser/eclipse/EclipseState/Schedule/WellConnections.cpp

```diff
@@ -29,6 +29,90 @@
 
 namespace Opm {
 
+namespace {
+
+    // Compute direction permutation corresponding to completion's
+    // direction.  First two elements of return value are directions
+    // perpendicular to completion while last element is direction
+    // along completion.
+    inline std::array< size_t, 3> directionIndices(const Opm::WellCompletion::DirectionEnum direction)
+    {
+        switch (direction) {
+        case Opm::WellCompletion::DirectionEnum::X:
+            return {{ 1,2,0 }};
+
+        case Opm::WellCompletion::DirectionEnum::Y:
+            return {{ 2,0,1}};
+
+        case Opm::WellCompletion::DirectionEnum::Z:
+            return {{ 9,1,2 }};
```

Comment:
That 9-value looks strange.

Reply:
Hehe - I am still in denial when it comes to the need for glasses:-)